libethcore/TransactionReceipt.h

```diff
@@ -55,11 +55,24 @@ class TransactionReceipt
 
     void streamRLP(RLPStream& _s) const;
 
-    void encode(bytes& receipt) const
+    void encode(bytes& receipt)
     {
+        if (m_receipt != bytes())
```

Comment:
this will make TransactionReceipt only can be called once.
if use cache, there should be a flag to indicate if the cache is valid